pkg/controller/shoot.go

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2019 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	gardencorev1alpha1 "github.com/gardener/gardener/pkg/apis/core/v1alpha1"
+	gardenv1beta1 "github.com/gardener/gardener/pkg/apis/garden/v1beta1"
+)
+
+// GetKubernetesVersion returns the Kubernetes version of the given Shoot.
+func GetKubernetesVersion(shoot *gardenv1beta1.Shoot) string {
```

Comment:
TBH, I would remove this function. It doesn't bring any benefit other than it's a little bit shorter. Access `shoot.Spec.Kubernetes.Version` directly.

Reply:
I believe it's still useful, as it abstracts away details of the `Shoot` structure. If they change, there would be a single place to fix.

Reply:
Hmm not sure, you still provide `*garden1beta1.Shoot` as an input, so I cannot throw `*gardencorev1alpha1.Shoot` into the function. Thus, when having a new shoot I would have to adapt this code anyways, or?

Reply:
OK, done.